lib/Conversion/ImportVerilog/Statements.cpp

```diff
@@ -463,6 +473,31 @@ struct StmtVisitor {
     return success();
   }
 
+  // Handle immediate assertion statements.
+  LogicalResult visit(const slang::ast::ImmediateAssertionStatement &stmt) {
+    auto cond = context.convertRvalueExpression(stmt.cond);
+    cond = context.convertToBool(cond);
+    if (!cond)
+      return failure();
+
+    auto defer = convertDeferAssert(stmt.isDeferred, stmt.isFinal);
+    switch (stmt.assertionKind) {
```

Comment:
Immediate assertions can have an action block associated with them, for example:
```
assert (x) $display("success");
assert (x) else $error("failure");
assert (x) $display("yay") else $error("nay");
```
It's not entirely clear to me if the presence of an action block simply turns the assert into an "if" statement, or whether the fact that it's an assert allows for additional global enabling/disabling controls to take effect. Do you know how this is supposed to work?

In any case, we can definitely deal with all of this later. If we postpone handling this, I'd suggest to add a check above the switch statement that ensures there is no action block associated with the assertion. That should produce an error like "unsupported immediate assertion action block" or similar.

Reply:
Yeah, I also noticed that. It's so similar to `if-else` in the slang AST. Therefore, I cannot make up my mind on how to deal with these different situations. It would be great if we could lower the situations you mentioned to `if-else`.

Reply:
That sounds great! 🥳